### DISCUSSION TITLE
CMake stuck at [33%]

### DISCUSSION THREAD

**Discussion by kajc10** [NVlabs/instant-ngp#340] [Q&A] [answered]
After issuing `cmake --build build --config RelWithDebInfo -j 16` the build process is stuck at Building CUDA object dependencies/tiny-cuda-nn/src/CMakeFiles/tiny-cuda-nn.dir/fully_fused_mlp.cu.o.

My setup:
CMake 3.22.3
CUDA 11.6
Python 3.8.12
gcc/g++ 7.5.0
Tesla V100 (4pcs)

**Comment by Tom94** [comment, marked as answer]
Hi there, the compilation times can be quite long: on the order of 15-30 minutes on older CPUs or even longer if too little RAM is available and swap space is hit. Have you tried letting it run for that long?

**Reply by kajc10** [reply]
Thanks, after 20 minutes it continued.
Hovewer, the build was not succesful.
`collect2: error: ld returned 1 exit status
CMakeFiles/testbed.dir/build.make:115: recipe for target 'testbed' failed
make[2]: *** [testbed] Error 1
CMakeFiles/Makefile2:199: recipe for target 'CMakeFiles/testbed.dir/all' failed
make[1]: *** [CMakeFiles/testbed.dir/all] Error 2
Makefile:90: recipe for target 'all' failed
make: *** [all] Error 2`

Should I open a new issues?

**Reply by Tom94** [reply]
Please feel free to open an additional discussion thread, yeah. Make sure to include all relevant information for others to help troubleshoot, i.e. the full compilation log, your operating system, GPU(s), compiler versions, and cmake version. (The above message doesn't contain any information about what actually went wrong other than that it's a linker error. :))

**Reply by collinabidi** [reply]
Any solutions to this? I've been having the same issue even with a lot of RAM.

**Reply by kajc10** [reply]
I thought the cmake build was stuck, but it turned out I just had to wait longer. (more than 20 mins)
